AppInspector.CLI/ResultsWriter.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.ApplicationInspector.Commands;
+using System;
+using System.IO;
+
+namespace Microsoft.ApplicationInspector.CLI
+{
+    /// <summary>
+    /// Wrapper for CLI only output arg validation which may be unique to a command 
+    /// and for allocating the correct writter type and format writter object 
+    /// </summary>
+    public static class ResultsWriter
+    {
+        public static void Write(TagDiffResult tagDiffResult, CLITagDiffCmdOptions options)
+        {
+            CommandResultsWriter writer = WriterFactory.GetWriter(options);
+            writer.WriteResults(tagDiffResult, options);
+            Finalize(writer, "Tag Diff");
+        }
+
+        public static void Write(TagTestResult tagTestResult, CLITagTestCmdOptions options)
+        {
+            CommandResultsWriter writer = WriterFactory.GetWriter(options);
+            writer.WriteResults(tagTestResult, options);
+            Finalize(writer, "Tag Test");
+        }
+
+        public static void Write(ExportTagsResult exportTagResult, CLIExportTagsCmdOptions options)
+        {
+            CommandResultsWriter writer = WriterFactory.GetWriter(options);
+            writer.WriteResults(exportTagResult, options);
+            Finalize(writer, "Export Tags");
+        }
+
+        public static void Write(VerifyRulesResult verifyRulesResult, CLIVerifyRulesCmdOptions options)
+        {
+            CommandResultsWriter writer = WriterFactory.GetWriter(options);
+            writer.WriteResults(verifyRulesResult, options);
+            Finalize(writer, "Verify Rules");
+        }
+
+        public static void Write(PackRulesResult packRulesResult, CLIPackRulesCmdOptions options)
+        {
+            CommandResultsWriter writer = WriterFactory.GetWriter(options);
+            writer.WriteResults(packRulesResult, options);
+            Finalize(writer, "Pack Rules");
+        }
+
+
+        /// <summary>
+        /// The only Write method that doesn't take a result object due to HTML needs which require more than 
+        /// </summary>
+        /// <param name="appProfile"></param>
+        /// <param name="options"></param>
+        public static void Write(AnalyzeResult analyzeResult, CLIAnalyzeCmdOptions options)
+        {
+            int MAX_HTML_REPORT_FILE_SIZE = 1024 * 1000 * 3;  //warn about potential slow rendering
+
+            //analyze with html format limit checks
+            if (options.OutputFileFormat == "html")
+            {
+                options.OutputFilePath = "output.html";
```

Comment:
What if the user set an output file path?

Reply:
Not supported currently as there are several dependencies on the local files for icons, js files etc.  requiring copying of the html directories to the target output path.  Future investigation to consolidate all files into a single file but not this PR.